tests/Integration/Behaviour/Features/Context/Domain/ProductFeatureContext.php

```diff
@@ -55,6 +62,191 @@ public function setProductTaxRulesGroup(string $taxRulesGroupReference, string $
         Cache::clean('product_id_tax_rules_group_*');
     }
 
+    /**
+     * @When I add product :productReference with following basic information:
+     *
+     * @param string $productReference
+     * @param TableNode $table
+     */
+    public function addProduct(string $productReference, TableNode $table): void
+    {
+        $data = $table->getRowsHash();
+
+        try {
+            $productId = $this->getCommandBus()->handle(new AddProductCommand(
+                $this->parseLocalizedArray($data['name']),
+                $this->getProductTypeValueByName($data['type'])
+            ));
+
+            $this->getSharedStorage()->set($productReference, $productId->getValue());
+        } catch (ProductException $e) {
+            $this->lastException = $e;
+        }
+    }
+
+    /**
+     * @Then /^product "(.+)" "(.+)" should be "(.+)"$/
```

Comment:
Since this assert function is dedicated to localized fields maybe it should be reflected in the Behat sentence
```suggestion
     * @Then /^product "(.+)" localized "(.+)" should be "(.+)"$/
```
for example

Reply:
okay :+1: